test/tests/functions/parsed/tests

```diff
@@ -1,5 +1,28 @@
 [Tests]
   design = 'source/functions/MooseParsedFunction.md'
+  [./nl_vars]
+    type = RunException
+    input = 'nl_vars_error.i'
+    cli_args = "Functions/func/vars='u v' Functions/func/vals='u v'"
+    expect_err = "The only variables supported by ParsedFunction are scalar variables, and var 'u' is not scalar."
+    requirement = "The system should error if a field variable is passed into a parsed function object, as this capability is currently not supported."
+    issues = '#15523'
+  [../]
+  [./nl_vars_allow_func_names]
```

Comment:
This looks good although this is the perfect case for using the "requirement detail" format. E.g we've created a syntax for where we have just a tiny variation on a requirement like you do for the second two cases here. You can factor out the common part of that sentence into a requirement and then have two `detail = <specialization>` lines. This helps keep our similar requirements together and shows that we still have test cases for these minor variations. It's worth getting in the habit of doing.

Reply:
Address this please. In general you should only refuse to address large items that will take you lots of time. It's not worth pushing back on the reviewers over 2 minutes of work :) 

Move the test to the end of the tests configuration file, and make a requirement/detail setup with the other error checking test.

Reply:
There was some confusion over these tests, and they were actually just removed, artifacts of an earlier commit idea. I will move it to the end of the list however.